src/content/reference/react/use-server.md

```diff
@@ -24,34 +24,233 @@ canary: true
+
+Server-side rendering and React Server Components are two features for optimizing performance of React apps and can complement one another.
+
+Server-side rendering (SSR) is the concept of taking a single render pass of a React app to resolve it to its base UI primitives, HTML elements for the web. This HTML is sent down to the client to serve as "scaffolding" while JavaScript is still being downloaded and evaluated.
+
+SRR is an optimization for ["time to paint"](https://developer.mozilla.org/en-US/docs/Glossary/First_paint) but it still requires all your app code to download and be evaluated on the client. The client then needs to [hydrate](/reference/react-dom/client/hydrateRoot#hydrating-server-rendered-html) this scaffolding to add interactivity.
+
+React Server Components (RSC) is the concept of breaking up your React [module dependency tree](/reference/react/use-client#marking-client-components-with-use-client) to determine what components can be rendered on the server and what can be sent to the client. In contrast to SSR, components that are rendered on the server have access to server APIs and can make data-fetches.
+
+These rendered Server Components are streamed down to the client in a special encoding to be translated to HTML or whatever client UI primitive. There is no hydration due to [limitations of Server Components](https://react-dev-git-use-client-fbopensource.vercel.app/reference/react/use-client#limitations).
+
+RSC is an optimization for ["time to interactive"](https://developer.mozilla.org/en-US/docs/Glossary/Time_to_interactive) as it reduces the code that needs to be downloaded and run by the client.
+
+</DeepDive>
+
+Here is the same username request form but rendered as a Server Component.
+
+```js
+// App.js
+
+export default App() {
+  function requestUsername(formData) {
+    'use server';
+    const username = formData.get('username');
+    // ...
+  }
+
+  return (
+    <form action={requestUsername}>
+      <input type="text" name="username" />
+      <button type="submit">Request</button>
+    </form>
+  );
+}
+```
+
+The `<form>` is rendered in a Server Component but the form submission will take place on the client. Before [hydration](/reference/react-dom/client/hydrateRoot#hydrating-server-rendered-html) or if JavaScript is disabled, this form will still allow users to call the server action.
+
+#### Handling return values {/*handling-return-values*/}
+
+You'll notice that in a username request form, there might be the chance that a username is not available. `requestUsername` should tell us if it fails or not. How does the form receive the return value of the server action?
+
+Here is where you can use [`useFormState`](/reference/react-dom/hooks/useFormState), a hook designed for receiving a server action return value that supports progressive enhancement.
+
+```js
+// App.js
+import {useFormState} from 'react';
+
+export default App() {
+  function requestUsername(formData) {
+    'use server';
```

Comment:
This is the first time in this doc we show a server action defined within the component. Should that be an example higher in the docs? The purpose being able to use a component prop or value from within the server action

Reply:
I've removed it for now and simplified the example usage -- at some point we should talk about this perhaps when we expand on closures